lang/py/src/avro/io.py

```diff
@@ -219,6 +232,41 @@ def read_double(self):
       ((ord(self.read(1)) & 0xffL) << 56))
     return STRUCT_DOUBLE.unpack(STRUCT_LONG.pack(bits))[0]
 
+  def read_decimal_from_bytes(self, precision, scale):
+    """
+    Decimal bytes are decoded as signed short, int or long depending on the
+    size of bytes.
+    """
+    size = self.read_long()
+    return self.read_decimal_from_fixed(precision, scale, size)
+
+  def read_decimal_from_fixed(self, precision, scale, size):
+    """
+    Decimal is encoded as fixed. Fixed instances are encoded using the
+    number of bytes declared in the schema.
+    """
+    datum = self.read(size)
+    unscaled_datum = 0
+    msb = struct.unpack('!b', datum[0])[0]
+    leftmost_bit = (msb >> 7) & 1
+    if leftmost_bit == 1:
+      modified_first_byte = ord(datum[0]) ^ (1 << 7)
+      datum = chr(modified_first_byte) + datum[1:]
+      for offset in range(size):
+        unscaled_datum <<= 8
+        unscaled_datum += ord(datum[offset])
+      unscaled_datum += pow(-2, (size*8) - 1)
+    else:
+      for offset in range(size):
+        unscaled_datum <<= 8
+        unscaled_datum += ord(datum[offset])
+
+    original_prec = getcontext().prec
+    getcontext().prec = precision
+    scaled_datum = Decimal(unscaled_datum) / Decimal(pow(10, scale))
```

Comment:
It looks like decimals have a [`scaleb`](https://docs.python.org/2/library/decimal.html#decimal.Decimal.scaleb) method for this:
```python
scaled_datum = Decimal(unscaled_data).scaleb(scale)
```